### PR TITLE
feat: OpenLineage extractor for databuilder

### PR DESCRIPTION
**Pull request by dechoma**
Signed-off-by: dechoma <dominik.choma@gmail.com>

### Summary of Changes

Databulder extractor to extract lineage information from OpenLineage format 
https://github.com/OpenLineage/OpenLineage

### CheckList
Make sure you have checked **all** steps below to ensure a timely review.
- [ ] PR title addresses the issue accurately and concisely. Example: "Updates the version of Flask to v1.0.2"
    - In case you are adding a dependency, check if the license complies with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
- [ ] PR includes a summary of changes.
- [ ] PR adds unit tests, updates existing unit tests, __OR__ documents why no test additions or modifications are needed.
- [ ] In case of new functionality, my PR adds documentation that describes how to use it.
    - All the public functions and the classes in the PR contain docstrings that explain what it does

### CODE REVIEW

**Comment by feng-tao** [comment]
@dechoma @mgorsk1 do you use openlineage in ING? If so, do you use to capture spark lineage or?

**Comment by mgorsk1** [comment]
we want to maintain openlineage compatible, in the first approach we are looking into spline lineage agent and mapping its output to openlineage format

File: databuilder/README.md [hunk 1]
```diff
@@ -860,6 +860,38 @@ Note that this provides accumulated view count which does [not effectively show
 
 If you are fine with `accumulated usage`, you could use TemplateVariableSubstitutionTransformer to transform Dict payload from [ModeDashboardUsageExtractor](./databuilder/extractor/dashboard/mode_analytics/mode_dashboard_usage_extractor.py) to fit [DashboardUsage](./docs/models.md#dashboardusage) and transform Dict to  [DashboardUsage](./docs/models.md#dashboardusage) by [TemplateVariableSubstitutionTransformer](./databuilder/transformer/template_variable_substitution_transformer.py), and [DictToModel](./databuilder/transformer/dict_to_model.py) transformers. ([Example](./databuilder/extractor/dashboard/mode_analytics/mode_dashboard_queries_extractor.py#L36) on how to combining these two transformers)
 
+#### [OpenLineageTableLineageExtractor](./databuilder/extractor/openlineage_extractor.py)
+A Extractor that extracts table lineage information from [OpenLineage](https://github.com/OpenLineage/OpenLineage) events.
+> :warning: Extractor expect input data in the form of **openLineage events in ndjson format**
```
Comment (mgorsk1) [inline databuilder/README.md:865]: expects*

File: databuilder/databuilder/extractor/openlineage_extractor.py [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+from databuilder.models.table_lineage import TableLineage
+
+LOGGER = logging.getLogger(__name__)
+
+
+class OpenLineageTableLineageExtractor(Extractor):
+    # Config keys
+    TABLE_LINEAGE_FILE_LOCATION = 'table_lineage_file_location'
+    CLUSTER_NAME = 'cluster_name'
+    OL_INPUTS_KEY = 'inputs_key'
```
Comment (mgorsk1) [inline databuilder/databuilder/extractor/openlineage_extractor.py:20]: can you describe (in README.md) what those `OL_` settings mean?

File: databuilder/databuilder/extractor/openlineage_extractor.py [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+                in_and_outs = ((inputs, outputs)
+                               for inputs in event[self.ol_inputs_key]
+                               for outputs in event[self.ol_outputs_key])
+                for row in in_and_outs:
+                    yield {'input': self._amundsen_dataset_key(row[0]),
+                           'output': self._amundsen_dataset_key(row[1])}
+            except KeyError:
+                LOGGER.error(f'Cannot extract valid input or output from Openlineage event \n {event} ')
+
+    def _amundsen_dataset_key(self, dataset: Dict) -> str:
```
Comment (mgorsk1) [inline databuilder/databuilder/extractor/openlineage_extractor.py:62]: we could describe what is the behavior of this method and that its default assumption is that it expects spark/hive table_name/path

**Review by mgorsk1 — CHANGES_REQUESTED** [review]

File: databuilder/databuilder/extractor/openlineage_extractor.py (hunk 2 above)
Comment (dechoma) [reply databuilder/databuilder/extractor/openlineage_extractor.py:22]: done

File: databuilder/README.md (hunk 1 above)
Comment (dechoma) [reply databuilder/README.md]: fixed

**Review by mgorsk1 — APPROVED** [review]

**Pull request merged**